pkg/networkservice/common/contexttranslation/client_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) 2020-2021 Doc.ai and/or its affiliates.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at:
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contexttranslation_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/utils/inject/injectconncontext"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/networkservicemesh/api/pkg/api/networkservice"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/contexttranslation"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/common/null"
+	"github.com/networkservicemesh/sdk/pkg/networkservice/core/next"
+)
+
+func TestConnectionContextTranslationClient(t *testing.T) {
+	capture := new(captureClient)
+
+	client := next.NewNetworkServiceClient(
+		contexttranslation.NewClient(),
+		capture,
+		injectconncontext.NewSetConnectionContextClient(),
+		null.NewClient(),
+	)
+
+	request := &networkservice.NetworkServiceRequest{
+		Connection: &networkservice.Connection{
+			Id: "id",
+			Context: &networkservice.ConnectionContext{
+				ExtraContext: map[string]string{
+					"test1": "testValue1",
+				},
+			},
+		},
+	}
+
+	// 1. Request
+
+	conn, err := client.Request(context.TODO(), request.Clone())
```

Comment:
I don't think `context.TODO` is suitable here.
A context with timeout, created at the start of the test, should probably be used here.